Arduino_package/hardware/variants/rtl8720dn_bw16/variant.h

```diff
@@ -0,0 +1,110 @@
+/*
+  Copyright (c) 2011 Arduino.  All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef _VARIANT_ARDUINO_AMEBA_
+#define _VARIANT_ARDUINO_AMEBA_
+
+#include "Arduino.h"
+#include <stdarg.h>
+
+#ifdef __cplusplus
+#include "LOGUARTClass.h"
+#include "UARTClassOne.h"
+#include "UARTClassTwo.h"
+extern "C"{
+#include "PinNames.h"
+}
+#endif
+
+#define MAIN_THREAD_STACK_SIZE        (4096 * 4)
+
+#define TOTAL_GPIO_PIN_NUM            (13)
+#define TOTAL_PWM_PIN_NUM             (4)
+
+#define LED_BUILTIN                   (10) //PA_14, Green
+#define LED_BUILTIN_G                 (10)
+#define LED_BUILTIN_R                 (12)
+#define LED_BUILTIN_B                 (11)
+
+#define FEATURE_ADC
+//#define FEATURE_DAC
+
+/* Analog pin mapping */
+#define A0                            4 //PB_1    3.3V    CH0
+#define A1                            5 //PB_2    3.3V    CH1
+#define A2                            6 //PB_3    3.3V    CH2
+
+/* DAC pin mapping */
+//#define DAC0    AD_3
+
+//#define SS                            10
+
+#define LOG_UART_MODIFIABLE_BAUD_RATE           1
+#define SERIAL_ONE_UART_MODIFIABLE_BAUD_RATE    1
+#define SERIAL_TWO_UART_MODIFIABLE_BAUD_RATE    1
+#define UART2_TX    PA_7
+#define UART2_RX    PA_8
+#define UART0_TX    PA_21    //PB_19  //Not available
+#define UART0_RX    PA_22    //PB_18  //Not available
+#define UART3_TX    PB_1    //PB_1
+#define UART3_RX    PB_2    //PB_2
+
+//All gpios. This is actrually necessary...
+#define PA7 0 
```

Comment:
I strongly suggest changing the pin Name as "D0" to "D12". "PA_7" is a pin name with a value and it will confuse with the "PA7" value.  "D1" should be fine and it is a symbol we have used for the Pin name of Arduino SDK.

Reply:
On the actrual dev board, there are no Dx labeling. What it have is three individual lable groups: GA0...GE2 and GA0...GA30 and PA7...PB3.   
Using Dx in code means let users search in pinmap every time using GPIO, quite painful!

Reply:
Yes, you are right. However,  "PA_7" and "PA7" are too similar, maybe another name but not "PA7". 
We are making a pin map for the board. It will help users a lot.